lib/bashcov/xtrace.rb

```diff
@@ -13,7 +15,10 @@ class Xtrace
     # @see http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in
     # @note We use a forward slash as delimiter since it's the only forbidden
     #   character in filenames on Unix and Windows.
-    PS4 = %Q{#{PREFIX}${BASH_SOURCE[0]}/${LINENO}: }
+    _READLINK = find_executable('readlink')
+    _GREADLINK = find_executable('greadlink')
+    READLINK = _GREADLINK.nil? ? _READLINK : _GREADLINK
+    PS4 = %Q{#{PREFIX}$(#{READLINK} -f ${BASH_SOURCE[0]})/${LINENO}: }
```

Comment:
Yeah, the `readlink` compatibility fiasco is a little annoying :/. Basically, the version of readlink shipped on OS X is not useful at all - the GNU version always prints the absolute path to the file, the BSD version prints the absolute path to the file only if the file was a symlink. It also doesn't have an option to follow symlinks.

Its possible to install the GNU version of coreutils on OS X, which I understand that most developers do, because the BSD equivalents are very lacking. They're all prefixed with a "g". Hence the reason why we'd find `greadlink`first and then find `readlink`. 

That said, after having read about `mkmf` and friends, I've decided to remove the readlink solution and replaced it with something that's a little slower than `readlink` but should work in our case -` $(cd $(dirname ${BASH_SOURCE[0]}); pwd)/$(basename ${BASH_SOURCE[0]})`. It won't follow symlinks, but it does get you the absolute path.